src/Raven.Server/Documents/Indexes/Static/CurrentIndexingScope.cs

```diff
@@ -133,6 +114,105 @@ public unsafe dynamic LoadDocument(LazyStringValue keyLazy, string keyString, st
             }
         }
 
+        public unsafe dynamic LoadCompareExchangeValue(LazyStringValue keyLazy, string keyString)
+        {
+            using (_loadCompareExchangeValueStats?.Start() ?? (_loadCompareExchangeValueStats = _stats?.For(IndexingOperation.LoadCompareExchangeValue)))
+            {
+                if (keyLazy == null && keyString == null)
+                    return DynamicNullObject.Null;
+
+                var source = Source;
+                var id = GetSourceId(source);
+
+                if (TryGetCompareExchangeKeySlice(keyLazy, keyString, out var keySlice) == false)
+                    return DynamicNullObject.Null;
+
+                Slice.From(_queryContext.Documents.Allocator, id, out var idSlice);
```

Comment:
Shouldn't be wrapped by `using` ?

Reply:
We cannot. It is used as a value in dictionary that is used at the end of the batch to write references. Will be released when indexing batch ends.

Reply:
OK